packages/table/src/table.tsx

```diff
@@ -506,6 +529,8 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         this.setState({
             columnWidths: newColumnWidths,
             focusedCell: enableFocus ? newFocusedCellCoordinates : undefined,
+            numFrozenColumnsClamped: this.getNumFrozenColumnsClamped(nextProps),
```

Comment:
I feel a little weird storing stuff in state that can be otherwise be derived from state/props. This is the whole "multiple sources of truth" problem. I'd much rather have a private function that just returns this number. I can be convinced otherwise -- but it's probably better to do that way.